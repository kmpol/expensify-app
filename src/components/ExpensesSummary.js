import React from 'react'
import { connect } from 'react-redux';
import selectedExpenses from '../selectors/expenses'
import selectedExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component {

    render() {
        const grammarExpenseDisplay = this.props.expenseCount === 1 ? "expense" : "expenses"
        return (
            <div>
                <p>Viewing {this.props.expenseCount} {grammarExpenseDisplay} with total: ${numeral(this.props.expensesInTotal / 100).format('0.00')}</p>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectedExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expensesInTotal: selectedExpensesTotal(visibleExpenses)
    }
}
export default connect(mapStateToProps)(ExpensesSummary)