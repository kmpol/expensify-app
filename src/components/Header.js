import { NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink activeClassName="is-active" to="/" exact={true}>Home page</NavLink>
        <NavLink activeClassName="is-active" to="/create">Create expense</NavLink>
        <NavLink activeClassName="is-active" to="/help">Help page</NavLink>
    </header>
);

export default Header;