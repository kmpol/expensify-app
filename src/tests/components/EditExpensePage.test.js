import React from 'react'
import toJSON from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import { EditExpensePage } from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'

configure({ adapter: new Adapter() });

let startEditExpense, startRemoveExpense, history, wrapper

beforeEach(() => {
    startEditExpense = jest.fn()
    startRemoveExpense = jest.fn()
    history = { push: jest.fn() }
    wrapper = shallow(<EditExpensePage startEditExpense={startEditExpense} startRemoveExpense={startRemoveExpense} history={history} expense={expenses[2]} />)
})

test('Should render Edit Expense Page', () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('Should edit expense correctly', () => {
    wrapper.find('ExpenseForm').prop('onSubmitJD')(expenses[2])
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(startEditExpense).toHaveBeenLastCalledWith(expenses[2].id, expenses[2])
})

test('Should remove expense correctly', () => {
    wrapper.find('button').simulate('click')
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(startRemoveExpense).toHaveBeenLastCalledWith({ id: expenses[2].id })
})