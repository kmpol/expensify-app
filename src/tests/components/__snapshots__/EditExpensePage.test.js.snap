// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Edit Expense Page 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 9999,
        "createdAt": 345600000,
        "description": "Credit card",
        "id": "3",
        "note": "",
      }
    }
    onSubmitJD={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
